core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -542,9 +549,7 @@ private[spark] class MesosClusterScheduler(
     tasks.foreach { case (offerId, taskInfos) =>
       driver.launchTasks(Collections.singleton(offerId), taskInfos.asJava)
     }
-    offers.asScala
-      .filter(o => !tasks.keySet.contains(o.getId))
-      .foreach(o => driver.declineOffer(o.getId))
+    currentOffers.asScala.filter(!_.used).foreach(o => driver.declineOffer(o.offerId))
```

Comment:
I think a for-comprehension would be slightly more efficient, though it's not a big deal.

```
for { o <- currentOffers.asScala if !o.used } driver.declineOffer(o.offerId)
```

`filter` creates an intermediate collection, while inside a for this can be fused into only one traversal.